python/nav/web/seeddb/utils/list.py

```diff
@@ -108,32 +101,49 @@ def _process_objects(query_set_values, value_list, edit_url=None,
     """
     # pick up which values are dictionaries and make note of their existing keys
     datakeys = defaultdict(set)
-    for obj in query_set_values:
+    for obj in queryset:
         for attr in value_list:
-            value = obj[attr]
+            value = _getattr(obj, attr)
             if isinstance(value, dict):
                 datakeys[attr].update(value)
     datakeys = dict([(k, list(sorted(v))) for k, v in datakeys.iteritems()])
 
     def _getvalues(obj):
         for attr in value_list:
-            value = obj[attr]
+            value = _getattr(obj, attr)
             if attr in datakeys:
                 for key in datakeys[attr]:
                     yield value.get(key, None)
             else:
                 yield value
 
-    objects = []
-    for obj in query_set_values:
+    rows = []
+    for obj in queryset:
         row = {
-            'pk': obj['pk'],
-            'values_list': list(_getvalues(obj))
+            'pk': obj.pk,
+            'values_list': list(_getvalues(obj)),
+            'model': obj,
         }
-        if edit_url:
-            row['url'] = reverse(edit_url, args=(obj[edit_url_attr],))
-        objects.append(row)
-    return objects, datakeys
+        if edit_url and edit_url_attr:
+            key = _getattr(obj, edit_url_attr)
+            row['url'] = reverse(edit_url, args=(key,))
+        rows.append(row)
+    return rows, datakeys
+
+
+def _getattr(obj, attr):
+    """Deep getattr for Django double underscore specs.
+
+    Should conform to the bassackwards ways of SeedDB at large.
+    """
+    try:
+        value = reduce(getattr, attr.split('__'), obj)
```

Comment:
Perhaps make it Python 3 compatible
http://python-future.org/compatible_idioms.html#reduce